### PR TITLE
Added Arbitrum gas results handling feature

### PR DESCRIPTION
**Pull request by kcharbo3**
## What is this PR doing?
Adds a feature to retrieve and compute gas costs for Arbitrum whenever the measure-gas.ts script is run for the Arbitrum testnet. This script will send a BLS tx and a normal token transfer tx to the Arbitrum testnet. Then, the contracts/scripts/util/arbitrum-gas-util.ts file is called to pull in gas prices from Arbitrum mainnet and append the results to the contracts/ArbitrumGasCosts.md file.

Also updates the measure-gas script to run with the new version of the BLSExpander contract.

## How can these changes be manually tested?
'npx hardhat run scripts/test/measure_gas.ts --network arbitrum_testnet' 
command should succeed. After it runs, check the contracts/ArbitrumGasCosts.md file to see if two new rows have been added to the table.

## Does this PR resolve or contribute to any issues?
No

## Checklist
- [X] I have manually tested these changes
- [X] Post a link to the PR in the group chat

## Guidelines
- If your PR is not ready, mark it as a draft
- The `resolve conversation` button is for reviewers, not authors
  - (But add a 'done' comment or similar)

### CODE REVIEW

File: contracts/package.json
Comment (jacque006) [inline contracts/package.json:21]: This (and a number of other dependencies) could be a `devDependencies` as it/they are not needed for contracts directly. @jzaki Any thoughts on separation of prod/dev dependencies for the contracts project? It's more relevant for projects where an npm package/runnable binary is produced and we want to keep it slim, but might be good for separation here as well.

File: contracts/scripts/test/measure_gas.ts
Comment (jacque006) [inline contracts/scripts/test/measure_gas.ts]: These vars are never reassigned and so can be `const`s.
```suggestion
    const actionsArr = [];
    const encodedFunctions = [];
```

File: contracts/scripts/util/arbitrum_gas_util.ts
Comment (jacque006) [inline contracts/scripts/util/arbitrum_gas_util.ts]: Non-blocking i/o ops preferred.
```
import { appendFile } from 'fs/promises';
...
```
```suggestion
    await fs.appendFile(ARBITRUM_GAS_COST_FILE, input);
```

File: contracts/scripts/util/arbitrum_gas_util.ts
Comment (jacque006) [inline contracts/scripts/util/arbitrum_gas_util.ts]: Should be able to use imports here.
```suggestion
import fs from 'fs';
/*
 Web3 needs to be used over Ethers since web3 tx receipts
 for Arbitrum correctly store the feeStats object.
 Ethers' tx receipts aren't updated to store that yet.
 */
import Web3 from 'web3';
import dotenv from 'dotenv';

dotenv.config();
```

**Review by jacque006 — COMMENTED** [review]
Nice work :+1: . Few nitpicks, but nothing that should hold this back.

Before we merge, could you please run the linter on the new/modified files.
```sh
# From project root
cd ./contracts
yarn lint --fix
```

Also @jzaki would be good to have you take a look as well.

File: contracts/package.json
Comment (jzaki) [reply contracts/package.json:21]: Good point. In the context of contracts, @openzeppelin/contracts is a dependency, and most others are `devDependencies`

Comment (jacque006) [reply contracts/package.json:21]: :+1: Moved to https://github.com/jzaki/bls-wallet/issues/139

**Comment by kcharbo3** [comment]
Ran the linter that made some changes. No errors for the new/modified files.

**Review by jacque006 — APPROVED** [review]

**Pull request merged**
